### ISSUE TITLE
VUID-1387 Misleading: vkDeviceCreateInfo suggests all required extensions must be in the device extension list

### ISSUE THREAD

**Issue by charles-lunarg** [KhronosGroup/Vulkan-Docs#1567]
For example:

> Missing extension required by the device extension VK_KHR_portability_subset: VK_KHR_get_physical_device_properties2. The Vulkan spec states: All required extensions for each extension in the VkDeviceCreateInfo::ppEnabledExtensionNames list must also be present in that list

VK_KHR_get_physical_device_properties2 is an instance extension, it must be enabled at instance creation time, thus wouldn't be in the device create info extension list.

The wording probably would best just assert that each extension must have all required extensions enabled and vulkan versions set, in case an extension needs 1.1 or 1.2.

**Comment by krOoze** [comment]
The VU is correct. The layers are misleading, or the MVK driver is non-conformant.

If a device extension has a device extension that depends on instance extension, and the extension is not enabled, then the driver must not list the extension as supported at all, and should result in `VK_ERROR_EXTENSION_NOT_PRESENT` and so VU is not needed for the case.

> Any device extension that has an instance extension dependency that is not enabled by `vkCreateInstance` is considered to be unsupported, hence it **must** not be returned by `vkEnumerateDeviceExtensionProperties` for any `VkPhysicalDevice` child of the instance.

**Comment by charles-lunarg** [comment]
> MVK driver is non-conformant
I mean, it is pretty well documented that MoltenVK isn't fully conformant, so that seems like the actual source of the error. 

Though, I can imagine an argument about 'discoverability of the portability extension' where if its not returned in the device extension list, a developer might assume it isn't supported or their MoltenVK install is buggy or out of date. 

I didn't know about this behavior and I've spent a fair amount of time reading & using Vullkan. I don't think its wrong to assume that others might have the same erroneous assumption. (That enumerate device extensions doesn't cull out unsupported extensions due to unmet dependencies). 

Regardless, thanks for the prompt reply, this issue should be raised in the MoltenVK repository

**Comment by krOoze** [comment]
Yes, it was to fix #659 and landed in 1.1.74

**Comment by r-potter** [comment]
The next spec update will have a minor text tweak to clarify that only dependencies on device extensions need to be included in VkDeviceCreateInfo::ppEnabledExtensionNames. 

The VU will become as follows (emphasis mine):

> All required **device** extensions for each extension in the `VkDeviceCreateInfo::pname:ppEnabledExtensionNames` list must: also be present in that list

Requirements on Core versions are deferred to the existing text on instance promotion.

krOoze is correct that implementations should not be enumerating device extensions when their instance level requirements aren't satisfied, and so it should not be possible to get into the situation where an enumerated device extension has a missing instance extension dependency, and to do so requires an invalid implementation. I've raised a portability issue for awareness on this.

The final question is whether the validation layers are behaving correctly in their reporting here. I took a look at the code that triggers this error, and my view is that it will behave correctly provided the implementation also behaves correctly (i.e. doesn't report device extensions that it shouldn't). Given that the purpose of the validation layers is to validate application behaviour, and not implementations, that seems sufficient.
